src/main/java/org/opensearch/jobscheduler/utils/JobDetailsService.java

```diff
@@ -33,31 +33,87 @@
 import org.opensearch.common.xcontent.LoggingDeprecationHandler;
 import org.opensearch.index.IndexNotFoundException;
 import org.opensearch.index.engine.DocumentMissingException;
+import org.opensearch.index.engine.Engine;
 import org.opensearch.index.engine.VersionConflictEngineException;
 import org.opensearch.index.seqno.SequenceNumbers;
+import org.opensearch.index.shard.IndexingOperationListener;
+import org.opensearch.index.shard.ShardId;
 import org.opensearch.jobscheduler.model.JobDetails;
 
 import java.nio.charset.StandardCharsets;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
 
-public class JobDetailsService {
+public class JobDetailsService implements IndexingOperationListener {
 
     private static final Logger logger = LogManager.getLogger(JobDetailsService.class);
-    private static final String JOB_DETAILS_INDEX_NAME = ".opensearch-plugins-job-details";
+    public static final String JOB_DETAILS_INDEX_NAME = ".opensearch-plugins-job-details";
     private static final String PLUGINS_JOB_DETAILS_MAPPING_FILE = "/mappings/opensearch_plugins_job_details.json";
 
     public static Long TIME_OUT_FOR_REQUEST = 10L;
     private final Client client;
     private final ClusterService clusterService;
+    private Set<String> indicesToListen;
 
-    public JobDetailsService(final Client client, final ClusterService clusterService) {
+    private static final ConcurrentMap<String, JobDetails> indexToJobDetails = IndexToJobDetails.getInstance();
+
+    public JobDetailsService(final Client client, final ClusterService clusterService, Set<String> indicesToListen) {
         this.client = client;
         this.clusterService = clusterService;
+        this.indicesToListen = indicesToListen;
     }
 
     public boolean jobDetailsIndexExist() {
         return clusterService.state().routingTable().hasIndex(JOB_DETAILS_INDEX_NAME);
     }
 
+    public static ConcurrentMap<String, JobDetails> getIndexToJobDetails() {
+        return JobDetailsService.indexToJobDetails;
+    }
+
+    private void updateIndicesToListen(String jobIndex) {
+        this.indicesToListen.add(jobIndex);
+    }
+
+    /**
+     * Adds a new JobDetails entry to the indexToJobDetails if jobType is null, or updates an existing JobDetails entry with the jobType
+     *
+     * @param extensionUniqueId the unique Id for the job details
+     * @param jobDetails the jobDetails to register
+     */
+    void updateIndexToJobDetails(String extensionUniqueId, JobDetails jobDetails) {
+        if (jobDetails.getJobType() != null && indexToJobDetails.containsKey(extensionUniqueId)) {
+            // Update existing JobDetails entry with job type
+            JobDetails existingJobDetails = indexToJobDetails.get(extensionUniqueId);
+            existingJobDetails.setJobType(jobDetails.getJobType());
```

Comment:
This means an extension could only register 1 job in a life time? 
Wouldn't that break when an extension wants to register multiple jobs?

Reply:
No, an extension will be able to register multiple jobs. The `indexToJobDetails` contains the initial configuration information for an extension's job to prepare the `JobSweeper` to begin listening to operations on the registered job index, such as the Job Index Name, Job Type, Job Runner Action name and Job Parameter Action name. This information does not change over time. Jobs are registered when extensions index a document to their corresponding job index.